EXPLORATORY DATA ANALYSIS

In [2]:
import pandas as pd

In [5]:
clean_df = pd.read_csv("superstore_clean.csv")
clean_df.head()

,Order ID,Order Date,Customer Name,Segment,Country,City,State,Region,Category,Sub-Category,Sales,Quantity,Discount,Profit,Year,Month,Month Name,Day Name
0,CA-2016-152156,2016-11-08,Claire Gute,Consumer,United States,Henderson,Kentucky,South,Furniture,Bookcases,261.9600,2,0.00,41.9136,2016,11,November,Tuesday
1,CA-2016-152156,2016-11-08,Claire Gute,Consumer,United States,Henderson,Kentucky,South,Furniture,Chairs,731.9400,3,0.00,219.5820,2016,11,November,Tuesday
2,CA-2016-138688,2016-06-12,Darrin Van Huff,Corporate,United States,Los Angeles,California,West,Office Supplies,Labels,14.6200,2,0.00,6.8714,2016,6,June,Sunday
3,US-2015-108966,2015-10-11,Sean O'Donnell,Consumer,United States,Fort Lauderdale,Florida,South,Furniture,Tables,957.5775,5,0.45,-383.0310,2015,10,October,Sunday
4,US-2015-108966,2015-10-11,Sean O'Donnell,Consumer,United States,Fort Lauderdale,Florida,South,Office Supplies,Storage,22.3680,2,0.20,2.5164,2015,10,October,Sunday


In [10]:
clean_df["Order Date"] = pd.to_datetime(
     clean_df["Order Date"],
     format="mixed"
)
clean_df

,Order ID,Order Date,Customer Name,Segment,Country,City,State,Region,Category,Sub-Category,Sales,Quantity,Discount,Profit,Year,Month,Month Name,Day Name
0,CA-2016-152156,2016-11-08,Claire Gute,Consumer,United States,Henderson,Kentucky,South,Furniture,Bookcases,261.9600,2,0.00,41.9136,2016,11,November,Tuesday
1,CA-2016-152156,2016-11-08,Claire Gute,Consumer,United States,Henderson,Kentucky,South,Furniture,Chairs,731.9400,3,0.00,219.5820,2016,11,November,Tuesday
2,CA-2016-138688,2016-06-12,Darrin Van Huff,Corporate,United States,Los Angeles,California,West,Office Supplies,Labels,14.6200,2,0.00,6.8714,2016,6,June,Sunday
3,US-2015-108966,2015-10-11,Sean O'Donnell,Consumer,United States,Fort Lauderdale,Florida,South,Furniture,Tables,957.5775,5,0.45,-383.0310,2015,10,October,Sunday
4,US-2015-108966,2015-10-11,Sean O'Donnell,Consumer,United States,Fort Lauderdale,Florida,South,Office Supplies,Storage,22.3680,2,0.20,2.5164,2015,10,October,Sunday
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
9988,CA-2014-110422,2014-01-21,Tom Boeckenhauer,Consumer,United States,Miami,Florida,South,Furniture,Furnishings,25.2480,3,0.20,4.1028,2014,1,January,Tuesday
9989,CA-2017-121258,2017-02-26,Dave Brooks,Consumer,United States,Costa Mesa,California,West,Furniture,Furnishings,91.9600,2,0.00,15.6332,2017,2,February,Sunday
9990,CA-2017-121258,2017-02-26,Dave Brooks,Consumer,United States,Costa Mesa,California,West,Technology,Phones,258.5760,2,0.20,19.3932,2017,2,February,Sunday
9991,CA-2017-121258,2017-02-26,Dave Brooks,Consumer,United States,Costa Mesa,California,West,Office Supplies,Paper,29.6000,4,0.00,13.3200,2017,2,February,Sunday


In [11]:
clean_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9993 entries, 0 to 9992
Data columns (total 18 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   Order ID       9993 non-null   object        
 1   Order Date     9993 non-null   datetime64[ns]
 2   Customer Name  9993 non-null   object        
 3   Segment        9993 non-null   object        
 4   Country        9993 non-null   object        
 5   City           9993 non-null   object        
 6   State          9993 non-null   object        
 7   Region         9993 non-null   object        
 8   Category       9993 non-null   object        
 9   Sub-Category   9993 non-null   object        
 10  Sales          9993 non-null   float64       
 11  Quantity       9993 non-null   int64         
 12  Discount       9993 non-null   float64       
 13  Profit         9993 non-null   float64       
 14  Year           9993 non-null   int64         
 15  Month          9993 n

In [14]:
# Calculate total sales and profit
total_sales = clean_df["Sales"].sum
total_profit = clean_df["Profit"].sum
total_sales , total_profit

(<bound method Series.sum of 0       261.9600
 1       731.9400
 2        14.6200
 3       957.5775
 4        22.3680
           ...   
 9988     25.2480
 9989     91.9600
 9990    258.5760
 9991     29.6000
 9992    243.1600
 Name: Sales, Length: 9993, dtype: float64>,
 <bound method Series.sum of 0        41.9136
 1       219.5820
 2         6.8714
 3      -383.0310
 4         2.5164
           ...   
 9988      4.1028
 9989     15.6332
 9990     19.3932
 9991     13.3200
 9992     72.9480
 Name: Profit, Length: 9993, dtype: float64>)

In [15]:
# Calculate average sales and profit
average_sales = clean_df["Sales"].sum()
average_profit = clean_df["Profit"].sum()
average_sales , average_profit

(2296919.4883000003, 286409.0805)

In [38]:
# Compare sales and profit together 
category_performance = clean_df.groupby(
     ["Category"]
)[["Sales" , "Profit"]].sum()
category_performance

,Sales,Profit
Category,,
Furniture,741718.4233,18463.3316
Office Supplies,719047.0320,122490.8008
Technology,836154.0330,145454.9481


In [18]:
# find most profitable category
category_performance["Profit"].idxmax() , category_performance["Profit"].max()

('Technology', 145454.9481)

In [21]:
# regional performance
regional_performance = clean_df.groupby(
     ["Region"]
)[["Sales" , "Profit"]].sum()
regional_performance

,Sales,Profit
Region,,
Central,501239.8908,39706.3625
East,678499.8680,91534.8388
South,391721.9050,46749.4303
West,725457.8245,108418.4489


In [23]:
# find most profitable region
regional_performance["Profit"].idxmax() , regional_performance["Sales"].max()

('West', 725457.8245)

In [25]:
# loss making transactions
loss_orders = clean_df[
     clean_df["Profit"] < 0
]
loss_orders

,Order ID,Order Date,Customer Name,Segment,Country,City,State,Region,Category,Sub-Category,Sales,Quantity,Discount,Profit,Year,Month,Month Name,Day Name
3,US-2015-108966,2015-10-11,Sean O'Donnell,Consumer,United States,Fort Lauderdale,Florida,South,Furniture,Tables,957.5775,5,0.45,-383.0310,2015,10,October,Sunday
14,US-2015-118983,2015-11-22,Harold Pawlan,Home Office,United States,Fort Worth,Texas,Central,Office Supplies,Appliances,68.8100,5,0.80,-123.8580,2015,11,November,Sunday
15,US-2015-118983,2015-11-22,Harold Pawlan,Home Office,United States,Fort Worth,Texas,Central,Office Supplies,Binders,2.5440,3,0.80,-3.8160,2015,11,November,Sunday
23,US-2017-156909,2017-07-16,Sandra Flanagan,Consumer,United States,Philadelphia,Pennsylvania,East,Furniture,Chairs,71.3720,2,0.30,-1.0196,2017,7,July,Sunday
27,US-2015-150630,2015-09-17,Tracy Blumstein,Consumer,United States,Philadelphia,Pennsylvania,East,Furniture,Bookcases,3083.4300,7,0.50,-1665.0522,2015,9,September,Thursday
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
9919,CA-2016-149272,2016-03-15,Muhammed Yedwab,Corporate,United States,Bryan,Texas,Central,Office Supplies,Binders,22.3860,7,0.80,-35.8176,2016,3,March,Tuesday
9920,CA-2014-111360,2014-11-24,Alyssa Tate,Home Office,United States,Akron,Ohio,East,Office Supplies,Binders,5.7420,3,0.70,-4.5936,2014,11,November,Monday
9930,CA-2015-104948,2015-11-13,Keith Herrera,Consumer,United States,San Bernardino,California,West,Furniture,Bookcases,683.3320,4,0.15,-40.1960,2015,11,November,Friday
9936,CA-2016-164889,2016-06-03,Christine Phan,Corporate,United States,Los Angeles,California,West,Furniture,Tables,71.0880,2,0.20,-1.7772,2016,6,June,Friday


In [26]:
loss_orders.shape[0]

1870

In [29]:
# check how many loss
loss_orders["Profit"].sum()   # this show negative result for absolute we use absolute 'abs'
abs(loss_orders["Profit"].sum())

156119.2269

In [30]:
# which category has the most loss
loss_by_category = loss_orders.groupby(
    "Category"
)["Profit"].sum()

loss_by_category

Category
Furniture         -60924.0502
Office Supplies   -56615.2585
Technology        -38579.9182
Name: Profit, dtype: float64

In [34]:
loss_by_category.idxmin() ,  loss_by_category.min()

('Furniture', -60924.0502)

TOP 10 CUSTOMERS

In [36]:
top_customers = clean_df.groupby("Customer Name")["Sales"].sum().sort_values(ascending=False).head(10)
top_customers

Customer Name
Sean Miller           25043.050
Tamara Chand          19052.218
Raymond Buch          15117.339
Tom Ashbrook          14595.620
Adrian Barton         14473.571
Ken Lonsdale          14175.229
Sanjit Chand          14142.334
Hunter Lopez          12873.298
Sanjit Engle          12209.438
Christopher Conant    12129.072
Name: Sales, dtype: float64

BUSINESS INSIGHTS

1) THE CATEGORY WITH HIGHEST PROFIT IS TECHNOLOGY.

2) THE REGION WITH HIGHEST PROFIT IS WEST REGION.

3) THERE ARE 1870 LOSS MAKING TRANSACTIONS, WITH FURNITURE CONTRIBUTING THE HIGHEST LOSS.

4) THE CUSTOMER GENERATING THE HIGHEST TOTAL SALES IS SEAN MILLER.